File: ender-does-backend/src/main/java/com/enderdev/enderdoesbackend/todo/models/Todo.java
package com.enderdev.enderdoesbackend.todo.models;

import com.enderdev.enderdoesbackend.user.models.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String title;

    private String body;

    private Boolean isDone;

    private LocalDateTime createdAt;

    private LocalDateTime completedAt;

    @ManyToOne
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;
}

File: ender-does-backend-NET/Todo/Models/Todo.cs
namespace ender_does_backend_NET.Todo.Models;

public class Todo
{
    public Guid Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Body { get; set; } = string.Empty;

    public bool IsDone { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? CompletedAt { get; set; }
    
    public Guid OwnerId { get; set; }

    public User.Models.User Owner { get; set; } = null!;
}